python/cudf/cudf/core/join/join.py

```diff
@@ -190,17 +191,25 @@ def perform_merge(self) -> cudf.DataFrame:
         )
 
         gather_kwargs = {
-            "nullify": True,
-            "check_bounds": False,
             "keep_index": self._using_left_index or self._using_right_index,
         }
         left_result = (
-            self.lhs._gather(gather_map=left_rows, **gather_kwargs)
+            self.lhs._gather(
+                ct.as_gather_map(
+                    left_rows, len(self.lhs), nullify=True, check_bounds=False
+                ),
+                **gather_kwargs,
+            )
```

Comment:
`_gather` now takes a `GatherMap`, so we must update callers everywhere. A bigger refactoring would slowly push these types into more places, but here I have just done the "closest" possible change.